### ISSUE TITLE
Failed to start in 6.8.1

### ISSUE THREAD

**Issue by mzch** [coralproject/talk#3282]
**Expected behavior:**
`yarn start` start Corel successfully

**Actual behavior:**
When `yarn start`, it failed to start. see below:

```node
talk@n23:~/talk$ yarn start
yarn run v1.22.5
$ NODE_ENV=production node dist/index.js
Error: Cannot find module '@hapi/joi'
Require stack:
- /srv/talk/talk/dist/core/server/config.js
- /srv/talk/talk/dist/core/server/services/i18n/index.js
- /srv/talk/talk/dist/core/server/errors/index.js
- /srv/talk/talk/dist/core/server/app/middleware/error.js
- /srv/talk/talk/dist/core/server/app/index.js
- /srv/talk/talk/dist/core/server/index.js
- /srv/talk/talk/dist/core/index.js
- /srv/talk/talk/dist/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:831:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:903:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/srv/talk/talk/src/core/server/config.ts:1:1)
    at Module._compile (internal/modules/cjs/loader.js:1015:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
    at Module.load (internal/modules/cjs/loader.js:879:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Module.require (internal/modules/cjs/loader.js:903:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/srv/talk/talk/src/core/server/services/i18n/index.ts:6:1)
    at Module._compile (internal/modules/cjs/loader.js:1015:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
    at Module.load (internal/modules/cjs/loader.js:879:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Related Issues:** 

**Versions:**

- NodeJS: v12.19.0
- NPM: 6.14.8
- MongoDB: 4.4.1
- Redis: 5.0.3
- OS: Debian 10.6

**Comment by cristiandean** [comment]
Hey @mzch, according to the documentation, [found here](https://docs.coralproject.net/coral/), the recommendation is to use `npm` and not `yarn`.

Anyway, did you run the `npm install` and `npm run build` command before `start`?

If the problem still persists, try deleting the node_modules directory and try again.

**Comment by mzch** [comment]
Oh, yes, I also did `npm start` before it. Well, I remove the existing repo and re-clone, ran `npm install` and `npm run build`,
then Corel runs successfully. Thanks.

**Comment by mzch** [comment]
Ah, I rebooted my VPS instance and tried to restart Coral, then failed.

```javascript
coral@n23:~/talk$ npm start

> @coralproject/talk@6.8.1 start
> NODE_ENV=production node dist/index.js

(node:30189) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
{"name":"coral","hostname":"n23","pid":30189,"level":30,"msg":"mongodb has connected","time":"2021-03-24T10:31:20.686Z","src":{"file":"/srv/coral/talk/src/core/server/services/mongodb/index.ts","line":42,"func":"createMongoDB"},"v":0}
{"name":"coral","hostname":"n23","pid":30189,"level":30,"queries":182,"msg":"loaded persisted queries","time":"2021-03-24T10:31:20.820Z","src":{"file":"/srv/coral/talk/src/core/server/graph/persisted/loader.ts","line":65,"func":"loadPersistedQueries"},"v":0}
{"name":"coral","hostname":"n23","pid":30189,"level":30,"listenerName":"analytics","msg":"listener was disabled","time":"2021-03-24T10:31:20.858Z","src":{"file":"/srv/coral/talk/src/core/server/events/publisher.ts","line":118,"func":"register"},"v":0}
{"name":"coral","hostname":"n23","pid":30189,"level":50,"err":{"message":"InProgressMigrationDetectedError: migration \"1579189174930\" was in progress","name":"InProgressMigrationDetectedError","stack":"Manager.pending (src/core/server/services/migrate/manager.ts:122:17)\nManager.executePendingMigrations (src/core/server/services/migrate/manager.ts:167:21)\nServer.process (src/core/server/index.ts:284:7)\nbootstrap (src/index.ts:54:5)\n","context":{"_id":"601bbf5fc36408dab77b6571","id":1579189174930,"clientID":"[Sensitive]","createdAt":"2021-02-04T09:33:19.191Z","state":"STARTED"}},"msg":"can not bootstrap server","time":"2021-03-24T10:31:20.954Z","src":{"file":"/srv/coral/talk/src/index.ts","line":59,"func":"bootstrap"},"v":0}
```

**Comment by wyattjoh** [comment]
It looks like you have an in progress migration. You'll need to remove that migration document from MongoDB (with the `{"id": 1579189174930}`) and re-run `npm start`.

**Comment by mzch** [comment]
I deleted the recored and tried again, then the following error occurred:

```
Mar 24 17:56:12 n23 npm[1335]: > @coralproject/talk@6.8.1 start
Mar 24 17:56:12 n23 npm[1335]: > NODE_ENV=production node dist/index.js
Mar 24 17:56:18 n23 npm[1335]: (node:1349) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
Mar 24 17:56:18 n23 npm[1335]: {"name":"coral","hostname":"n23","pid":1349,"level":30,"msg":"mongodb has connected","time":"2021-03-25T00:56:18.460Z","src":{"file":"/srv/coral/talk/src/core/server/services/mongodb/index.ts","line":42,"func":"createMongoDB"},"v":0}
Mar 24 17:56:18 n23 npm[1335]: {"name":"coral","hostname":"n23","pid":1349,"level":30,"queries":182,"msg":"loaded persisted queries","time":"2021-03-25T00:56:18.572Z","src":{"file":"/srv/coral/talk/src/core/server/graph/persisted/loader.ts","line":65,"func":"loadPersistedQueries"},"v":0}
Mar 24 17:56:18 n23 npm[1335]: {"name":"coral","hostname":"n23","pid":1349,"level":30,"listenerName":"analytics","msg":"listener was disabled","time":"2021-03-25T00:56:18.605Z","src":{"file":"/srv/coral/talk/src/core/server/events/publisher.ts","line":118,"func":"register"},"v":0}
Mar 24 17:56:18 n23 npm[1335]: {"name":"coral","hostname":"n23","pid":1349,"level":30,"pending":3,"msg":"executing pending migrations","time":"2021-03-25T00:56:18.645Z","src":{"file":"/srv/coral/talk/src/core/server/services/migrate/manager.ts","line":178,"func":"executePendingMigrations"},"v":0}
Mar 24 17:56:18 n23 npm[1335]: {"name":"coral","hostname":"n23","pid":1349,"migrationName":"indexes","migrationID":1579189174930,"level":30,"msg":"starting index migration","time":"2021-03-25T00:56:18.675Z","src":{"file":"/srv/coral/talk/src/core/server/services/migrate/manager.ts","line":200,"func":"executePendingMigrations"},"v":0}
Mar 24 18:02:18 n23 npm[1335]: {"name":"coral","hostname":"n23","pid":1349,"collectionName":"migrations","level":50,"err":{"message":"connection 0 to 192.168.207.10:27017 timed out","name":"MongoNetworkError","stack":"Socket.<anonymous> (node_modules/mongodb/lib/core/connection/connection.js:355:7)\n"},"indexSpec":{"id":1},"indexOptions":{"unique":true},"msg":"could not create index","time":"2021-03-25T01:02:18.810Z","src":{"file":"/srv/coral/talk/src/core/server/services/migrate/indexing.ts","line":48,"func":"createIndex"},"v":0}
Mar 24 18:02:18 n23 npm[1335]: {"name":"coral","hostname":"n23","pid":1349,"level":50,"err":{"message":"connection 0 to 192.168.207.10:27017 timed out","name":"MongoNetworkError","stack":"Socket.<anonymous> (node_modules/mongodb/lib/core/connection/connection.js:355:7)\n"},"msg":"can not bootstrap server","time":"2021-03-25T01:02:18.822Z","src":{"file":"/srv/coral/talk/src/index.ts","line":59,"func":"bootstrap"},"v":0}
Mar 24 18:02:18 n23 systemd[1]: coral.service: Main process exited, code=exited, status=1/FAILURE
Mar 24 18:02:18 n23 systemd[1]: coral.service: Failed with result 'exit-code'.
```

- .env
```
MONGODB_URI=mongodb://coral:password@192.168.207.10:27017/coral?keepAlive=true&poolSize=30&autoReconnect=true&socketTimeoutMS=360000&connectTimeoutMS=360000
```

**Comment by wyattjoh** [comment]
It seems like the migration process exceeded the timeout, I'd suggest setting a higher timeout for the connection and retrying!

**Comment by mzch** [comment]
I tried `socketTimeoutMS=720000` parameter but the same result. Does migration process need more than 12 min?

**Comment by wyattjoh** [comment]
It depends on how large your dataset is! If there are a lot of comments, and/or the resources on the MongoDB instances doesn't have a lot of resources, it may take a while to perform the upgrade.

**Comment by mzch** [comment]
I created a new database, no comments, and at first Coral started normally, but once shut down it and restart Coral, then the above error happened. What causes an error?

**Comment by mzch** [comment]
Fixed this issue in 6.9.0. Thanks!
